dpnp/dpnp_algo/dpnp_algo.pxd

```diff
@@ -200,13 +204,29 @@ cdef extern from "dpnp_iface.hpp":
 # C function pointer to the C library template functions
 ctypedef void(*fptr_1in_1out_t)(void * , void * , size_t)
 ctypedef void(*fptr_2in_1out_t)(void * , void*, void*, size_t)
+ctypedef void(*fptr_2in_1out_2sz_t)(const void *, const void *, const void *, const size_t, const size_t)
 ctypedef void(*fptr_blas_gemm_2in_1out_t)(void * , void * , void * , size_t, size_t, size_t)
 ctypedef void(*dpnp_reduction_c_t)(const void *, const size_t, void * , const long*, const size_t, const long*, const size_t, const void * , const long*)
 
 cdef dparray call_fptr_1in_1out(DPNPFuncName fptr_name, dparray x1, dparray_shape_type result_shape)
 cdef dparray call_fptr_2in_1out(DPNPFuncName fptr_name, dparray x1, dparray x2, dparray_shape_type result_shape)
 
 
+ctypedef fused dpnp_numeric:
```

Comment:
Fused types is a bit scaring for me. I played with them sometime ago and reject this solution (select "void*" instead).
Did you try `complex<double>` and bool in this "fused type" approach? Does it work?

Reply:
Yes both `complex<double>` and `bool` work fine. Currently I don't see alternative way to use a cython function like a c++ function template.